code/script.js

```diff
@@ -222,104 +226,130 @@ const generateBoard = () => {
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+      alert(`Sí, the person has a ${value}! Keep all people that have a ${value}.`)
+    } else { 
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+      alert(`Nooo, the person does not have a ${value}.. Remove all people that have a ${value}.`)
+    }
+
+  } else if (category === 'hair') {
+    if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      alert(`Yes, the person has ${value} hair, YAY! Keep all people with ${value} hair.`)
+    } else {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      alert(`Unfortunately, the person does not have ${value} hair. Remove all people with ${value} hair.`)
+    }
 
+  } else if (category === 'eyes') {
+    if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      alert(`Yes, the person has ${value} eyes, WIHO! Keep all people with ${value} eyes.`)
+    } else {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      alert(`Sorry, the person does not have ${value} eyes. Remove all people with ${value} eyes.`)
+    }
+  }
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard(keep)
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  // Stores the interaction from the player in a variable.
+  let thePlayerGuess = confirm(`Would you like to guess on ${personToConfirm}?`)
+ 
+  if (thePlayerGuess) {
+    checkMyGuess(personToConfirm)
+  } else {
+    alert("Please make another guess!")
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+   // 2. Set a Message to show in the win or lose section accordingly
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `CONGRATS! ${secret.name} is the correct guess! Great job, well done!`
+  } else {
+    winOrLoseText.innerHTML = `I'm sorry, that was incorrect. The right answer is ${secret.name}!`
+  }
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex'
   // 4. Hide the game board
+  boardWrapper.style.display = 'none'
```

Comment:
It would have been so esay to added a bit of sound effects here, just 2 line of code,